src/main/groovy/de/dkfz/execution/jobs/JobManager.java

```diff
@@ -0,0 +1,174 @@
+        C c = createCommand(job, job.jobName, job.getListOfProcessingCommand(), tool, job.getParameters(), dependencies, job.arrayIndices);
+        c.setJob(job);
+        return c;
+    }
+/*
+    public Command.DummyCommand createDummyCommand(Job job, String jobName, List<String> arraySettings) {
+        return new Command.DummyCommand(job,jobName, arraySettings != null && arraySettings.size() > 0);
+    }
+*/
+    public abstract JobDependencyID createJobDependencyID(Job job, String jobResult);
+
+    public abstract ProcessingCommands convertResourceSet(AppConfig configuration, ResourceSet resourceSet);
+
+    public abstract ProcessingCommands parseProcessingCommands(String alignmentProcessingOptions);
+
+    public abstract ProcessingCommands getProcessingCommandsFromConfiguration(AppConfig configuration, String toolID);
+
+    public abstract ProcessingCommands extractProcessingCommandsFromToolScript(File file);
+
+    public List<C> getListOfCreatedCommands() {
+        List<C> newList = new LinkedList<>();
+        synchronized (listOfCreatedCommands) {
+            newList.addAll(listOfCreatedCommands);
+        }
+        return newList;
+    }
+
+    /**
+     * Tries to reverse assemble job information out of an executed command.
+     * The format should be [id], [command, i.e. qsub...]
+     *
+     * @param commandString
+     * @return
+     */
+    public abstract Job parseToJob(String commandString);
+
+    public abstract GenericJobInfo parseGenericJobInfo(String command);
+
+    public abstract JobResult convertToArrayResult(Job arrayChildJob, JobResult parentJobsResult, int arrayIndex);
+
+    /**
+     * Queries the status of all jobs in the list.
+     *
+     * @param jobIDs
+     * @return
+     */
+    public abstract Map<String, JobState> queryJobStatus(List<String> jobIDs);
+
+    public abstract void queryJobAbortion(List<Job> executedJobs,  ExecutionService executionService);
+
+    public abstract void addJobStatusChangeListener(Job job);
+
+    public abstract String getLogFileWildcard(Job job);
+
+    public abstract boolean compareJobIDs(String jobID, String id);
+
+    public void addCommandToList(C pbsCommand) {
+        synchronized (listOfCreatedCommands) {
+            listOfCreatedCommands.add(pbsCommand);
+        }
+    }
+
+    public int waitForJobsToFinish() {
+        return 0;
+    }
+
+    public void addSpecificSettingsToConfiguration(AppConfig configuration) {
+
+    }
+
+    /**
+     * Tries to get the log for a running job.
+     * Returns an empty array, if the job's jobState is not RUNNING
+     *
+     * @param job
+     * @return
+     */
+    public abstract String[] peekLogFile(Job job);
+
+    /**
+     * Stores a new job jobState info to an de.dkfz.execution contexts job jobState log file.
+     *
+     * @param job
+     */
+    /**
+     * Stores a new job jobState info to an execution contexts job jobState log file.
+     *
+     * @param job
+     */
```

Comment:
Haha, looks like a merge problem. ExecutionContext objects are a central part of Roddy. JobState contains info about the status of a job.